src/include/cuda_math_helper.h

```diff
@@ -1052,6 +1053,14 @@ HD_FUNC double rnorm4d(double a, double b, double c, double d) {
   OTHER_FUNC_FP(T, T2, T3, T4)               \
   ABS_FUNC(T, T2, T3, T4)                    \
   MAKE_FUNC(T, T2, T3, T4)
+#else
+#define DEFINE_FUNC_FOR_FLOAT(T, T2, T3, T4) \
```

Comment:
This change is done because HIP headers already include the operators for int# and float#, so redefining them would cause compiler errors.